contrib/endpoints/src/api_manager/api_manager_impl.h

```diff
@@ -73,6 +79,9 @@ class ApiManagerImpl : public ApiManager {
 
   // A weighted service selector.
   std::unique_ptr<WeightedSelector> service_selector_;
+
+  // Access control to service_context_map_
+  std::mutex service_context_mutex_;
```

Comment:
Most of api-manager code are not supposed to run in multiple thread environment.  Nginx is single threaded.  We need to do a lot of code hardening if it is moved to multiple threaded environment. We need a overall design to make api-manager thread safe.